lib/ansible/plugins/inventory/foreman.py

```diff
@@ -121,134 +126,139 @@ def _get_session(self):
             self.session.verify = self.get_option('validate_certs')
         return self.session
 
-    def _get_json(self, url, ignore_errors=None):
-
-        if not self.use_cache or url not in self._cache.get(self.cache_key, {}):
+    def _get_all_hosts(self, url):
 
-            if self.cache_key not in self._cache:
+        if self.use_cache:
+            try:
+                inventory = self._cache[self.cache_key][url]
```

Comment:
Shouldn't this be using `.get()`
```suggestion
                inventory = self._cache.get(self.cache_key)[url]
```

Reply:
No, because the interface for self._cache has been made more dict-like. We should get a KeyError for expired cache keys, but a dictionary's get() method should not cause a KeyError.